symbolic-symcache/src/writer.rs

```diff
@@ -416,13 +430,25 @@ where
         Ok(index)
     }
 
-    fn insert_lines(
+    /// Takes an iterator of [`LineInfo`]s and returns a vector containing [`LineRecord`](format::LineRecord)s
+    /// for those lines whose address is between `start_address` and `end_address`.
+    ///
+    /// - If the difference between the addresses of two consecutive
+    /// lines `L1` and `L2` is greater than 255, dummy line records with the same file and line
+    /// information as L1 will be inserted between the two.
+    ///
+    /// - One call of this function will
+    /// produce a maximum of 2^16 line records and will not produce line records with an address more than
+    /// 2^16 bytes after the start address. If either of these limits is exceeded, the function will return
+    /// early with the address of the first line that could not be processed; it is then up to
+    /// the caller to call it again with that address as the new start address.
+    fn make_line_records(
```

Comment:
This now reminds me a little of iterator's `take` and `take_while` functions. Suggestion, what if we call this `take_line_records` or `take_lines`?